moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp

```diff
@@ -301,18 +301,18 @@ void RobotStateDisplay::newRobotStateCallback(const moveit_msgs::DisplayRobotSta
 {
   if (!robot_model_)
     return;
-  if (!robot_state_)
-    robot_state_.reset(new robot_state::RobotState(robot_model_));
+
   // possibly use TF to construct a robot_state::Transforms object to pass in to the conversion function?
   try
   {
-    robot_state::robotStateMsgToRobotState(state_msg->state, *robot_state_);
+    auto robot_state = std::make_shared<robot_state::RobotState>(robot_model_);
+    robot_state::robotStateMsgToRobotState(state_msg->state, *robot_state);
+    robot_state_.swap(robot_state);
```

Comment:
Yes, this makes sense. However, I preferred to skip the allocation because the failure case is very uncommon but the allocation is required on every update.
How about using `setToDefaultValues` to enforce a safe state in case of error?